### PR TITLE
Create mysql-network-policy

### PR DESCRIPTION
**Pull request by jdorescf**
First draft of tutorial access giving example of how to connect to an arbitrary TCP/UDP application.

As per the discussion here:
https://jira.cfdata.org/browse/PMM-866

### CODE REVIEW

File: content/cloudflare-one/tutorials/mysql-network-policy.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+---
+updated: 2024-03-11
+category: 🔐 Zero Trust
+pcx_content_type: tutorial
+title: Access and secure a MySQL database using cloudflare tunnel and network policies
```
Comment (crwaters16) [inline content/cloudflare-one/tutorials/mysql-network-policy.md]:
```suggestion
title: Access and secure a MySQL database using Cloudflare Tunnel and network policies
```

File: content/cloudflare-one/tutorials/mysql-network-policy.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+---
+
+# Access and secure a MySQL database using cloudflare tunnal and network policies
```
Comment (crwaters16) [inline content/cloudflare-one/tutorials/mysql-network-policy.md]:
```suggestion
# Access and secure a MySQL database using Cloudflare Tunnel and network policies
```

File: content/cloudflare-one/tutorials/mysql-network-policy.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+Using Cloudflare Tunnel's private networks, users can connect to arbitrary non-browser based TCP/UDP applications, such as, databases. Furthermore, it is possible to set up Network Policies that implement zero trust controls to define who and what can access those applications using the WARP client.
```
Comment (crwaters16) [inline content/cloudflare-one/tutorials/mysql-network-policy.md]:
```suggestion
Using Cloudflare Tunnel's private networks, users can connect to arbitrary non-browser based TCP/UDP applications, like databases. You can set up network policies that implement zero trust controls to define who and what can access those applications using the WARP client.
```

File: content/cloudflare-one/tutorials/mysql-network-policy.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+By the end of this tutorial, users that pass network policies will be able to access a remote MySQL database available through a Cloudflare tunnel on TCP port 3306.
```
Comment (crwaters16) [inline content/cloudflare-one/tutorials/mysql-network-policy.md]:
```suggestion
By the end of this tutorial, users that pass network policies will be able to access a remote MySQL database available through a Cloudflare Tunnel on TCP port 3306.
```

File: content/cloudflare-one/tutorials/mysql-network-policy.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+Make sure you have:
+
+- A MySQL database listening for remote connections and configured with users that can connect remotely
+- [Resolver policies](/cloudflare-one/policies/gateway/resolver-policies/) enabled on your account (optional)
+
+{{</tutorial-prereqs>}}
+
+{{<tutorial-step title="Create a Cloudflare Tunnel">}}
+
+First, install `cloudflared` on a server in your private network. This server should have connectivity to the MySQL database.
```
Comment (crwaters16) [inline content/cloudflare-one/tutorials/mysql-network-policy.md]:
```suggestion
Install `cloudflared` on a server in your private network. This server should have connectivity to the MySQL database.
```

File: content/cloudflare-one/tutorials/mysql-network-policy.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+2. Select **Save tunnel**.
+
+The application and (optionally) DNS server are now connected to Cloudflare.
+
+{{</tutorial-step>}}
+
+{{<tutorial-step title="Create a Gateway network policy">}}
+
+1. Go to **Gateway** > **Firewall Policies** > **Network**.
+2. Add a [network policy](/cloudflare-one/policies/gateway/network-policies/) that targets the private IP address and the port of the MySQL database (by default port 3306). The example below allows access to the database to the users that enrolled into WARP using an `@example.com` email address. The network policies can also take into consideration [device posture checks](/cloudflare-one/identity/devices/).
```
Comment (crwaters16) [inline content/cloudflare-one/tutorials/mysql-network-policy.md]:
```suggestion
2. Add a [network policy](/cloudflare-one/policies/gateway/network-policies/) that targets the private IP address and the port of the MySQL database (port 3306 by default). The following example allows access to the database to the users that enrolled into WARP using an `@example.com` email address. The network policies can also take into consideration [device posture checks](/cloudflare-one/identity/devices/).
```

File: content/cloudflare-one/tutorials/mysql-network-policy.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+| Selector | Operator | Value | Logic | Action |
+| -------- | -------- | ----- | ----- | ------ |
+| Destination IP | in | `10.128.0.175` | And | Allow |
+| Destination Port | in | `3306` | And | |
+| User Email | matches regex | `.*example.com` | |
+
+In addition to the Allow rule above, it is recommended to add a [catch-all block policy to the bottom of your network policy list to enforce a default-deny model](learning-paths/replace-vpn/build-policies/).
```
Comment (crwaters16) [inline content/cloudflare-one/tutorials/mysql-network-policy.md]:
```suggestion
In addition to the Allow rule above, Cloudflare recommends adding a [catch-all block policy](learning-paths/replace-vpn/build-policies/) to the bottom of your network policy list to enforce a default-deny model.
```

File: content/cloudflare-one/tutorials/mysql-network-policy.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    | -------- | -------- | ---------------------- |
+    | Domain     | in       | `internalrecord.com` |
+
+4. In **Select DNS resolver**, select _Configure custom DNS resolvers_.
+5. Enter the private IP address of your DNS server.
+6. In the dropdown menu, select _`<IP-address> - Private`_.
+7. (Optional) Enter a custom port.
+8. Select **Create policy**.
+
+If your internal DNS server has an A record for the MySQL database, then users will be able to connect to the server using this record. For example, assuming a BIND server that includes the entry:
```
Comment (crwaters16) [inline content/cloudflare-one/tutorials/mysql-network-policy.md]:
```suggestion
If your internal DNS server has an `A` record for the MySQL database, users can connect to the server using this record. For example, assuming a BIND server that includes the entry:
```

File: content/cloudflare-one/tutorials/mysql-network-policy.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+`mysql IN  A  10.128.0.175`
+
+then allowed WARP users can connect to the MySQL database at `mysql.internalrecord.com` using the MySQL client of their choice.
```
Comment (crwaters16) [inline content/cloudflare-one/tutorials/mysql-network-policy.md]:
```suggestion
Allowed WARP users can connect to the MySQL database at `mysql.internalrecord.com` using the MySQL client of their choice.
```

File: content/cloudflare-one/tutorials/mysql-network-policy.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+By the end of this tutorial, users that pass network policies will be able to access a remote MySQL database available through a Cloudflare Tunnel on TCP port 3306.
+
+{{<tutorial>}}
+
+{{<tutorial-prereqs>}}
+
+Make sure you have:
+
+- A MySQL database listening for remote connections and configured with users that can connect remotely
+- [Resolver policies](/cloudflare-one/policies/gateway/resolver-policies/) enabled on your account (optional)
```
Comment (crwaters16) [inline content/cloudflare-one/tutorials/mysql-network-policy.md]:
```suggestion
- (Optional)[Resolver policies](/cloudflare-one/policies/gateway/resolver-policies/) enabled on your account 
```

File: content/cloudflare-one/tutorials/mysql-network-policy.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+{{<render file="tunnel/_create-tunnel.md" productFolder="cloudflare-one">}}
+
+{{</tutorial-step>}}
+
+{{<tutorial-step title="Add private network routes">}}
+
+1. In the **Private Networks** tab, add the following IP addresses:
+
+  - Private IP/CIDR of your MySQL server (for example, `10.128.0.175/32`)
+  - Private IP/CIDR of your internal DNS server (optional)
```
Comment (crwaters16) [inline content/cloudflare-one/tutorials/mysql-network-policy.md]:
```suggestion
  - (Optional) Private IP/CIDR of your internal DNS server
```

File: content/cloudflare-one/tutorials/mysql-network-policy.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+2. Select **Save tunnel**.
+
+The application and (optionally) DNS server are now connected to Cloudflare.
```
Comment (crwaters16) [inline content/cloudflare-one/tutorials/mysql-network-policy.md]:
```suggestion
The application and (optional) DNS server are now connected to Cloudflare.
```

File: content/cloudflare-one/tutorials/mysql-network-policy.md [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+| Destination Port | in | `3306` | And | |
+| User Email | matches regex | `.*example.com` | |
+
+In addition to the Allow rule above, Cloudflare recommends adding a [catch-all block policy](/learning-paths/replace-vpn/build-policies/) to the bottom of your network policy list to enforce a default-deny model.
+
+Allowed WARP users can now connect to the MySQL server at `10.128.0.175` using the MySQL client of their choice.
+
+{{</tutorial-step>}}
+
+{{<tutorial-step title="Create a Gateway resolver policy (optional)">}}
```
Comment (crwaters16) [inline content/cloudflare-one/tutorials/mysql-network-policy.md]:
```suggestion
{{<tutorial-step title="(Optional) Create a Gateway resolver policy">}}
```

File: content/cloudflare-one/tutorials/mysql-network-policy.md [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+
+To allow users to access the MySQL database using an internal hostname instead of the private IP address, configure a Gateway resolver policy.
+
+1. Go to **Gateway** > **Resolver policies**.
+2. Select **Add a policy**.
+3. Create an expression to match against the private [domain](/cloudflare-one/policies/gateway/resolver-policies/#domain) or [hostname](/cloudflare-one/policies/gateway/resolver-policies/#host) of the application:
```
Comment (crwaters16) [inline content/cloudflare-one/tutorials/mysql-network-policy.md]:
```suggestion
3. Create an expression to match against the private [domain](/cloudflare-one/policies/gateway/resolver-policies/#domain) or [hostname](/cloudflare-one/policies/gateway/resolver-policies/#host) of the application, like in the following example:
```

**Review by crwaters16 — APPROVED** [review]

**Review by jdorescf — COMMENTED** [review]
Looks good to me, thanks for the review

**Pull request merged**